scripts/run_mcmc.py

```diff
@@ -0,0 +1,295 @@
+
+    # check if init tree is provided
+    if params.init_tree_fp is None:
+        # infer dimensions of tree from data
+        n_mutations, m_cells = mut_mat.shape
+        # split rng key
+        rng_tree, rng = random.split(rng, 2)
+        #  generate random trees (uniform sampling) as adjacency matrix
+        #  / add +1 for root
+        tree = tree_inf.generate_random_tree(rng_tree, n_nodes=n_mutations + 1)
+        tree = jnp.array(tree)
+        # make Tree
+        labels = jnp.arange(n_mutations + 1)
+        init_tree = tree_inf.Tree(tree, labels)
+        logging.info("Generated random tree.")
+
+    else:
+        # parse tree from file given as input
+
+        # make path and check if Path exists
+        p = Path(params.init_tree_fp)
+        if not p.exists():
+            raise FileNotFoundError(f"File {params.init_tree_fp} does not exist.")
+
+        if params.init_TreeNode:
+            init_tree_node = serialize.read_tree_node(params.init_tree_fp)
+            # convert TreeNode to Tree
+            init_tree = tree_inf.tree_from_tree_node(init_tree_node)
+            logging.info("Loaded tree (TreeNode) from file.")
+
+        elif params.init_tree_mcmc_no is not None:
+            mcmc_sample = serialize.read_mcmc_samples(params.init_tree_fp)[
+                params.init_tree_mcmc_no - 1
+            ]
+            _, init_tree, _ = tree_inf.unpack_sample(mcmc_sample)
+            logging.info("Loaded tree (mcmc sample) from file.")
+        else:
+            raise ValueError(
+                "Please provide either TreeNode or mcmc sample number,"
+                " to read in tree from file."
+            )
+
+    # Make Move Probabilities
+    prune_and_reattach = config["move_probs"]["prune_and_reattach"]
+    swap_node_labels = config["move_probs"]["swap_node_labels"]
+    swap_subtrees = config["move_probs"]["swap_subtrees"]
+    move_probs = MoveProbabilities(prune_and_reattach, swap_node_labels, swap_subtrees)
+
+    # run mcmc sampler
+    mcmc_sampler(
+        rng_key=rng,
+        data=mut_mat,
+        error_rates=(config["fpr"], config["fnr"]),
+        move_probs=move_probs,
+        num_samples=config["num_samples"],
+        num_burn_in=config["burn_in"],
+        output_dir=Path(params.out_dir),
+        thinning=config["thinning"],
+        init_tree=init_tree,
+        timestamp=kwargs.get("timestamp", None),
+    )
+
+    print("Done!")
+
+
+def get_config(config_fp: str) -> dict:
+    """Load the config file.
+
+    Args:
+        config_fp: str
+            Filepath of the config file.
+    Returns:
+        config: dict
+            config file as dict
+    """
+    # load config from file to json object
+    with open(config_fp, "r") as f:
+        config = json.load(f)
+
+    return config
+
+
+def main() -> None:
+    """
+    Main function.
+    """
+    # Parse command line arguments
+    params = create_parser()
+
+    # TODO: check params
```

Comment:
This is one of the reasons why I love [Pydantic](https://docs.pydantic.dev/latest/) :slightly_smiling_face: